### PR TITLE
Enable automerge for new version PRs of community packages

### PR DESCRIPTION
**Pull request by ringods**
## Description

Fixes: #4424

* Enables automerge also for new version PRs of community packages.
* Refactored duplicate steps from 2 workflows in a separate composite action

### CODE REVIEW

**Comment by ringods** [comment]
Triggered a manual run, a new version of `runpod-native` community package was picked up, but the execution of the workflow failed because I made a mistake in the path to the composite action: `./github/...` instead of `./.github/...`

https://github.com/pulumi/registry/actions/runs/9594431540/job/26457015261

**Comment by ringods** [comment]
My second attempt was not yet succesfull, but the third one was:

https://github.com/pulumi/registry/actions/runs/9599696212

This run found 1 community package which had a new version:

| Job Creating PR | Automerged PR |
|--------|--------|
| [pulumiverse/pulumi-acme@v0.1.1](https://github.com/pulumi/registry/actions/runs/9599696212/job/26474133011) | #4781 |

This was the most I could test this PR.

**Review by cnunciato — APPROVED** [review]
One question that needn't necessarily block, but would prefer unless objection.

File: .github/workflows/generate-package-metadata.yml [hunk 1]
```diff
@@ -75,21 +75,15 @@ jobs:
...
         with:
-          token: ${{ secrets.PULUMI_BOT_TOKEN }}
-          committer: Pulumi Bot <bot@pulumi.com>
-          author: Pulumi Bot <bot@pulumi.com>
-          commit-message: "Publish Package Metadata for ${{ env.PROVIDER_SHORT_NAME }}@${{ env.PROVIDER_VERSION }}"
-          title: "Publish Package Metadata ${{ env.PROVIDER_SHORT_NAME }}@${{ env.PROVIDER_VERSION }}"
-          body: ""
-          branch: "${{ env.PROVIDER_SHORT_NAME }}/${{ github.run_id }}-${{ github.run_number }}"   
-  
+          GITHUB_TOKEN: ${{ secrets.PULUMI_BOT_TOKEN }}
```
[the remark this answers is not in the document]
Comment (ringods) [reply .github/workflows/generate-package-metadata.yml]: @cnunciato fair point. I updated the code from `GITHUB_TOKEN` to `github_token`. Using `token` doesn't make clear it is a Github token.

Composite actions indeed don't inherit secrets by default:

> If you want to pass a secret to a composite action, you need to do it explicitly as an input.

Source: [`secrets` context](https://docs.github.com/en/actions/learn-github-actions/contexts#secrets-context)

**Comment by sean1588** [comment]
thank you @ringods !!!

**Pull request merged**